src/perl/lib/wtsi_clarity/epp/generic/step_assigner_by_field.pm

```diff
@@ -0,0 +1,212 @@
+Readonly::Scalar my $NEXT_ACTION_PATH           => q(/stp:actions/next-actions/next-action);
+Readonly::Scalar my $SAMPLE_URI_PATH            => q{/art:artifact/sample/@uri};
+Readonly::Scalar my $PROCEED_TO_SEQUENCING_PATH => q{/smp:sample/udf:field[@name="WTSI Proceed To Sequencing?"]};
+## use critic
+
+Readonly::Array my @PROCEED_VALUES => qw{Y YES};
+
+extends 'wtsi_clarity::epp';
+
+our $VERSION = '0.0';
+
+# properties needed to configure the script
+
+has 'step_url' => (
+  isa        => 'Str',
+  is         => 'ro',
+  required   => 1,
+);
+
+has 'field_name' => (
+  isa        => 'Str',
+  is         => 'ro',
+  required   => 1,
+);
+
+has 'next_step_name' => (
+  isa        => 'Str',
+  is         => 'ro',
+  required   => 1,
+);
+
+sub _next_action_uri {
+  my ($self) = @_;
+
+  return join q{/}, ($self->step_url, 'actions');
+}
+
+has '_nextActionsList' => (
+  isa             => 'XML::LibXML::Document',
+  is              => 'rw',
+  required        => 0,
+  lazy_build      => 1,
+);
+sub _build__nextActionsList {
+  my $self = shift;
+
+  return $self->fetch_and_parse($self->_next_action_uri);
+}
+
+has '_current_protocol_step' => (
+  isa             => 'XML::LibXML::Document',
+  is              => 'rw',
+  required        => 0,
+  lazy_build      => 1,
+);
+sub _build__current_protocol_step {
+  my $self = shift;
+
+  return $self->fetch_and_parse($self->_nextActionsList->findvalue($CONFIGURATION_URI_PATH));
+}
+
+has '_transition_step_uri_by_step_name' => (
+  isa             => 'HashRef',
+  is              => 'rw',
+  required        => 0,
+  lazy_build      => 1,
+);
+sub _build__transition_step_uri_by_step_name {
+  my $self = shift;
+
+  my $transitions = ();
+  $transitions->{'current_step'} = $self->_nextActionsList->findvalue($CONFIGURATION_URI_PATH);
+
+  my @transition_nodes = $self->_current_protocol_step->findnodes($TRANSITION_URI_PATH)->get_nodelist();
+
+  foreach my $transition_node (@transition_nodes) {
+    if ($transition_node->getAttribute(q{name}) eq $self->next_step_name) {
+      $transitions->{$self->next_step_name} = $transition_node->getAttribute(q{next-step-uri});
+    }
+  }
+
+  return $transitions;
+}
+
+sub _get_artifact_by_uri {
+  my ($self, $artifact_uri) = @_;
+
+  return $self->fetch_and_parse($artifact_uri);
+}
+
+sub _get_sample_by_uri {
+  my ($self, $sample_uri) = @_;
+
+  return $self->fetch_and_parse($sample_uri);
+}
+
+sub _set_next_actions {
+  my $self = shift;
+
+  my @next_action_nodes = $self->_nextActionsList->findnodes($NEXT_ACTION_PATH)->get_nodelist();
+
+  foreach my $next_action_node (@next_action_nodes) {
+    my $artifact_xml = $self->_get_artifact_by_uri($next_action_node->getAttribute(q{artifact-uri}));
+    my $sample_xml = $self->_get_sample_by_uri($artifact_xml->findvalue($SAMPLE_URI_PATH));
+    my $is_proceed_to_sequencing = $sample_xml->findvalue($PROCEED_TO_SEQUENCING_PATH);
```

Comment:
Yes, you are right. I forgot to change it to the field_name in the UDF field XPATH. I'll do it now.